colossalai/auto_parallel/solver/sharding_strategy.py

```diff
@@ -37,20 +40,36 @@ class ShardingStrategy:
     input_shardings: List[ShardingSpec] = None
 
 
+class OperandType(Enum):
+    """
+    An operand can come from the argument list of an operator or the parameter list of a module.
+    """
+    ARG = 0
+    PARAM = 1
+
+
+@dataclass
+class Operand:
+    name: str
+    type: OperandType
+
+
 @dataclass
-class TrainCycleItem:
+class TrainCycleCost:
     """
     TrainCycleItem is a dataclass to store the items which have different values for the forward and backward pass
     in a training iteration.
 
     Args:
-        fwd (Any): the item for the forward pass
-        bwd (Any): the item for the backward pass
-        total (Any): the total value for the forward and backward pass
+        fwd (float): the item for the forward pass
+        bwd (float): the item for the backward pass
     """
```

Comment:
I prefer the previous version, because the fwd cost may be a tuple for memory cost(activation memory cost, weight memory cost). And the communication cost and computation cost can benefit from this. For example, the weight memory cost, not the total cost, is just what we need to compute the communication cost of batch sharding cases.

Reply:
Ok I see

Reply:
recovered.